base/shared/src/main/scala/scalaz/typeclass/DebugSyntax.scala

```diff
@@ -2,9 +2,25 @@ package scalaz
 package typeclass
 
 import language.experimental.macros
+import language.implicitConversions
 
 trait DebugSyntax {
   implicit final class ToDebugOps[A: Debug](self: A) {
     def debug: String = macro meta.Ops.f_0
   }
+
+  implicit final def debugInterpolator(sc: StringContext): DebugInterpolator.DebugInterpolator =
+    DebugInterpolator.DebugInterpolator(sc)
+}
+
+object DebugInterpolator {
+  final case class Debugs(override val toString: String) extends AnyVal
```

Comment:
arguably this could be (private constructor; not a case class)

```scala
final class Debugged private (override val toString: String) extends AnyVal
object Debugged {
  implicit def mat[A](x: A)(implicit A: Debug[A]): Debugged = 
    new Debugged(A.debug(a))
}
```

but I don't think it really matters

Reply:
I renamed this class to "HasDebug" because it makes the type error much more useful:

```scala
scala> z"${bar}"
<console>:20: error: type mismatch;
 found   : Bar
 required: scalaz.typeclass.DebugInterpolator.HasDebug
       z"${bar}"
           ^
```

Seems more intuitive to require `HasDebug` than `Debugs`. (Although I wish I could customize this somehow and say "Need Debug". But I don't think this is possible here.)

Reply:
You can, [actually](https://scalafiddle.io/sf/Uycxpb8/0):

```scala
// Start writing your ScalaFiddle code here

trait Show[A] {
  def shows(a: A): String
}

final class Shewn(override val toString: String)
object Shewn extends Shewn0 {
  implicit def materialize[A](a: A)(implicit A: Show[A]): Shewn =
    new Shewn(A shows a)
}
sealed abstract class Shewn0 {
  @annotation.implicitAmbiguous("Cannot use the `z` interpolator to interpolate a value of type ${A}, as no implicit Show[${A}] instance is in scope.")
  implicit def fail1[A](a: A): Shewn = ???
  implicit def fail2[A](a: A): Shewn = ???
}

final class Z(sc: StringContext) {
  def z(shewns: Shewn*): String = sc.s(shewns : _*)
}

implicit def conv(sc: StringContext): Z = new Z(sc)

implicit val showStr: Show[String] = _.toString

println(z"asdf ${"zyfg"} xjrl") // compiles
println(z"this is an int ${1}")
// ScalaFiddle.scala:27: error: Cannot use the `z` interpolator to interpolate a value of type scala.this.Int, as no implicit Show[scala.this.Int] instance is in scope.
//  println(z"this is an int ${1}")
//                             ^
```

(I think I'll backport this to 7.3.x)

Reply:
Oh, I like this a lot, let me update mine.